habitat/tasks/rearrange/multi_task/composite_task.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+import os.path as osp
+from typing import Any, Dict, List, Optional
+
+import magnum as mn
+import numpy as np
+import yaml
+
+from habitat.core.dataset import Episode
+from habitat.core.registry import registry
+from habitat.tasks.rearrange.multi_task.pddl_domain import PddlDomain
+from habitat.tasks.rearrange.multi_task.rearrange_pddl import (
+    Action,
+    Predicate,
+    RearrangeObjectTypes,
+    SetState,
+    parse_func,
+)
+from habitat.tasks.rearrange.rearrange_task import RearrangeTask
+from habitat.tasks.rearrange.utils import logger
+
+
+@registry.register_task(name="RearrangeCompositeTask-v0")
+class CompositeTask(RearrangeTask):
+    """
+    All tasks using a combination of sub-tasks (skills) should utilize this task.
+    """
+
+    def __init__(self, *args, config, dataset=None, **kwargs):
+        super().__init__(config=config, *args, dataset=dataset, **kwargs)
+
+        task_spec_path = osp.join(
+            self._config.TASK_SPEC_BASE_PATH, self._config.TASK_SPEC + ".yaml"
+        )
+
+        with open(task_spec_path, "r") as f:
+            task_def = yaml.safe_load(f)
+
+        # Stores configuration for the task.
+        self.task_def: Dict[str, Any] = task_def
```

Comment:
I am not sure how to be more specific since this refers to any element that might be in the configuration dictionary such as a list, another dict, int, or string.